features/steps/steps.py

```diff
@@ -221,3 +249,41 @@ def step_impl(context, related, relating, other_entity, condition):
                         errors.append(instance_structure_error(inst, rel.RelatingObject))
 
     handle_errors(context, errors)
+
+@then('The relative placement of that {entity} must be provided by an {other_entity} entity')
+def step_impl(context, entity, other_entity):
+    if getattr(context, 'applicable', True):
+        errors = []
+        for obj in context.instances:
+            if not obj.ObjectPlacement.is_a(other_entity):
+                errors.append(instance_placement_error(obj, other_entity, "", "", "", ""))
+        handle_errors(context, errors)
+
+
+@then(
+    'The {entity} entity must point to the {other_entity} of the container element established with {relationship} relationship')
+def step_impl(context, entity, other_entity, relationship):
+    if getattr(context, 'applicable', True):
+        errors = []
+
+        stmt_to_op = {'IfcRelAggregates': 'Decomposes'}
+        assert relationship in stmt_to_op
+
+        for entity in context.instances:
+            entity_relation = getattr(entity, stmt_to_op[relationship])[0]
+            if relationship == 'IfcRelAggregates':
```

Comment:
Why do we not follow the same approach as above:

```
    for rel in relationships:
        if rel.RelatingObject ...
```

Reply:
Changed. Additionally relationship attributes are derived from the csv file now, not hardcoded.